src/app/utils/factories/filters.ts

```diff
@@ -0,0 +1,71 @@
+import { Coffee } from "@shared/models/classes/coffee.class";
+import { FilterType } from "@shared/models/types/filter-type.type";
+import { Merchandise } from "@shared/models/classes/merchandise.class";
+import { Product } from "@shared/models/classes/product.class";
+import { ProductType } from "@app/shared/models/types/product-type.type";
+
+export function getFilterableProperties(products: Product[], productType: ProductType): FilterType {
```

Comment:
Without a factory function, code was getting spread out all over the place in kind of an ugly way. Getting to this factory function is what took me the most time. Hopefully, it should now be the case that all of the code related to products and their filterable properties is in the one place, instead of across several files.

In addition to this, I wasn't sure which properties you wanted to filter on, so I just added all of them. The good thing about this solution, is that you can add and remove what you want from here, and you shouldn't need to worry about changing anything anywhere else. It should (hopefully) just work.

Reply:
This is a good way to think about it. 

However, instead of using a generic factory function that switches on productType, we can use the classes that are extended based on productType as there is already the discinction.

I have updated product-option modifiers ticket and will update this ticket as well